### ISSUE TITLE
Add support for CREATE TYPE .. AS ENUM

### ISSUE THREAD

**Issue by JanecekPetr** [jOOQ/jOOQ#7570]
### Expected behavior and actual behavior:

The following DDL for creating an enumerated type on Postgres should be supported by the jOOQ parser (I thought https://github.com/jOOQ/jOOQ/issues/7386 might do this?):

```sql
CREATE TYPE FOO AS ENUM ('BAR', 'BAZ', 'QUX');
```

but instead it throws

```
GENERATOR, GLOBAL TEMPORARY TABLE, INDEX, OR ALTER VIEW, OR REPLACE VIEW, SCHEMA, SEQUENCE, TABLE, TEMPORARY TABLE, UNIQUE INDEX, or VIEW expected:
[1:8] CREATE [*]TYPE FOO AS ENUM ('BAR', 'BAZ', 'QUX');
```

### Steps to reproduce the problem (include example code if possible):

Run the DDL abode through the jOOQ parser.

### Versions:

- jOOQ: 3.11
- Java: N/A
- Database (include vendor): The original DDL has been written for and is being accepted by Postgres.
- OS: N/A
- JDBC Driver (include name if inofficial driver): N/A

**Comment by lukaseder** [comment]
Thanks for your report and for taking the time to create these.

Indeed, we should eventually add support for `CREATE TYPE` statements, including `CREATE TYPE .. AS ENUM`. This could probably be emulated in other databases by remembering the type information and adding `CHECK` constraints where applicable.

Once that's supported, parser support will be obvious.

**Comment by lukaseder** [comment]
Dropping types (#7774) and altering types (#7775) should be supported as well

**Comment by lukaseder** [comment]
This is implemented for jOOQ 3.12
